test/parallel_api/numeric/numeric.ops/in_place_scan.pass.cpp

```diff
@@ -19,64 +19,331 @@
 
 #include "support/test_config.h"
 #include "support/utils.h"
+#include "support/scan_serial_impl.h"
 
 #if TEST_DPCPP_BACKEND_PRESENT
-#    include "support/utils_sycl.h"
-#    include "support/sycl_alloc_utils.h"
-#    include "support/scan_serial_impl.h"
+#include <CL/sycl.hpp>
+#endif
+using namespace TestUtils;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+static const char kMsgInclusiveScanNormal [] = "Wrong effect from inclusive scan (non-inplace)";
+static const char kMsgInclusiveScanInplace[] = "Wrong effect from inclusive scan (inplace)";
+static const char kMsgExclusiveScanNormal [] = "Wrong effect from exclusive scan (non-inplace)";
+static const char kMsgExclusiveScanInplace[] = "Wrong effect from exclusive scan (inplace)";
+
+// TODO: replace data generation with random data and update check to compare result to
+// the result of a serial implementation of the algorithm
+template <typename Iterator1, typename Size>
+void
+initialize_data(Iterator1 host_keys, Size n)
+{
+    const Size kStartVal = 1;
+    for (Size i = 0; i != n; ++i)
+        host_keys[i] = kStartVal + i;
+}
+
+struct TestingAlgoritmInclusiveScan
```

Comment:
Lets keep just one such class instead of four.
You can pass F (to class constructor or directly to call_onedpl) as a callable parameter and call line this:
```
template<typename F>
struct invoker
{
    F f;
    template <typename ...Args>
    void operator()(Args&&... args)
    {
        f(::std::forward<Args>(args)...);
    }
};
```